File: frontend/src/app/configuration/application-entities/application-entity.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { ApplicationEntity } from './application-entity.model';
import { API_URL } from 'app/config';

@Injectable({
  providedIn: 'root'
})
export class ApplicationEntityService {

  constructor(
    private http:HttpClient
  ) { }

  retrieveApplicationEntities(page,size) {
    return this.http.get<ApplicationEntity[]>(`${API_URL}/application-entities?page=${page}&size=${size}`);
  }

  deleteApplicationEntity(id){
    return this.http.delete(`${API_URL}/application-entities/${id}`);
  }

  retrieveApplicationEntity(id){
    return this.http.get<ApplicationEntity>(`${API_URL}/application-entities/${id}`);
  }

  updateApplicationEntity(id, applicationEntity){
    return this.http.put(
          `${API_URL}/application-entities/${id}`
                , applicationEntity);
  }

  createApplicationEntity(applicationEntity){
    return this.http.post(
              `${API_URL}/application-entities`
                , applicationEntity);
  }


  startApplicationEntity(id){
    return this.http.put(`${API_URL}/application-entities/${id}/start`, null);
  }

  stopApplicationEntity(id){
    return this.http.put(`${API_URL}/application-entities/${id}/stop`, null);
  }
}

File: backend/src/main/java/com/almustkbal/pacs/services/ApplicationEntityService.java
package com.almustkbal.pacs.services;

import org.springframework.data.domain.Pageable;

import com.almustkbal.pacs.entities.ApplicationEntity;

import java.util.List;

import org.springframework.data.domain.Page;

public interface ApplicationEntityService {

	List<ApplicationEntity> getAllApplicationEntities();

	Page<ApplicationEntity> getApplicationEntities(Pageable pageable);

	ApplicationEntity getApplicationEntityById(long id);

	ApplicationEntity createApplicationEntity(ApplicationEntity ApplicationEntity);

	ApplicationEntity updateApplicationEntity(long id, ApplicationEntity ApplicationEntity);

	void deleteApplicationEntity(long id);

	void startApplicationEntity(long id);

	void stopApplicationEntity(long id);
}
